docs/guide/custom-functions.md

```diff
@@ -140,16 +220,72 @@ the calculations.
 // arguments here are displayed just to show the structure
 public hyper(ast, state) {
     return 'Hyperformula'.length;
+  };
+```
+
+### `runFunction()`
+
+To test your function's configuration, run the built-in `runFunction()` function.
+
+The `runFunction()` function:
+- Validates your function's [optional parameter](#optional-parameters) settings
+- Validates your [argument validation options](#argument-validation-options)
+- Checks if values returned by your function are in the right format
+
+```javascript
+import { FunctionPlugin } from 'hyperformula';
+
+export class CountHF extends FunctionPlugin {
+
+  public static implementedFunctions = {
+    'HYPER': {
+      method: 'hyper',
+      // set your optional parameters
+      arrayFunction: false,
+      // set your argument validation options
+      parameters: {
+        passSubtype: false
+      }
+    }
+  };
+
+  public hyper(ast, state) {
+    this.runFunction(HYPER())
+    
+    return 'Hyperformula'.length
   }
 };
 ```
 
+### Returning errors
+
+If you want your custom function to return an error, check the [API reference](../api) for the HyperFormula [error types](types-of-errors.md).
+
+For example, if you want to return a `#DIV/0!` error, use the [`CellError` class](../api/classes/cellerror.md), and the [DIV_BY_ZERO](../api/enums/errortype.md#div-by-zero) error type:
+
+```javascript
+// import `CellError` and `ErrorType`
+import { FunctionPlugin, CellError, ErrorType } from "hyperformula";
+
+public hyper({ args }) {
+    if (!args.length) {
+      // create a `CellError` instance with an `ErrorType` of `DIV_BY_ZERO`
+      return new CellError(ErrorType.DIV_BY_ZERO);
+    }
+    
+    else {
+      return 'Hyperformula'.length;
+    }
+  };
+```
+
+The error displays as `#DIV/0`, and gets properly translated.
+
```

Comment:
I've just noticed that my suggestion removed the optional error message. Maybe we should mention that all error types support optional string message that we encourage to give users more details on the reasons for error and how they can avoid it?

Also, here is a good place to mention that errors will be returned by ie getCellValue methods but wrapped in DetailedCellError type that will translate those errors according to i18n settings. I think that would be a full 101 on errors then

Reply:
@wojciechczerniak - done: 11d9bed